### PR TITLE
US016: ParkingSlot Edit Action 

### PR DESCRIPTION
**Pull request by iamurtazo**

### CODE REVIEW

File: ParkingZoneApp.Tests/Controller/ParkingSlotControllerTests.cs [hunk 1]
```diff
@@ -148,4 +150,98 @@ public void GivenCreateModel_WhenCreateIsCalled_ThenModelStateIsTrueAndReturnsRe
...
+
+        //Act
+        var result = _controller.Edit(1);
+
+        //Assert
+        var notFoundResult = Assert.IsType<NotFoundResult>(result);
+        Assert.Equal(404, notFoundResult.StatusCode);
+        Assert.NotNull(result);
+        _slotService.Verify(x => x.GetById(1), Times.Once);
+
```
Comment (SamDotNetDev) [inline ParkingZoneApp.Tests/Controller/ParkingSlotControllerTests.cs]: space

File: ParkingZoneApp.Tests/Controller/ParkingSlotControllerTests.cs [hunk 2]
```diff
@@ -148,4 +150,98 @@ public void GivenCreateModel_WhenCreateIsCalled_ThenModelStateIsTrueAndReturnsRe
...
+        Assert.NotNull(model);
+        _slotService.Verify(x => x.GetById(1), Times.Once);
+
+    }
+
+    [Fact]
+    public void GivenIdAndEditViewModel_WhenEditIsCalled_ThenReturnsNotFoundResult()
+    {
+        //Arrange
+
```
Comment (SamDotNetDev) [inline ParkingZoneApp.Tests/Controller/ParkingSlotControllerTests.cs:194]: space

File: ParkingZoneApp.Tests/Controller/ParkingSlotControllerTests.cs [hunk 3]
```diff
@@ -148,4 +150,98 @@ public void GivenCreateModel_WhenCreateIsCalled_ThenModelStateIsTrueAndReturnsRe
...
+        var result = _controller.Edit(1);
+        var model = ((ViewResult)result).Model;
+
+        //Assert
+        Assert.IsType<ViewResult>(result);
+        Assert.Equal(JsonSerializer.Serialize(model), JsonSerializer.Serialize(editVM));
+        Assert.NotNull(result);
+        Assert.NotNull(model);
+        _slotService.Verify(x => x.GetById(1), Times.Once);
+
```
Comment (SamDotNetDev) [inline ParkingZoneApp.Tests/Controller/ParkingSlotControllerTests.cs]: space

File: ParkingZoneApp.Tests/Controller/ParkingSlotControllerTests.cs [hunk 4]
```diff
@@ -148,4 +150,98 @@ public void GivenCreateModel_WhenCreateIsCalled_ThenModelStateIsTrueAndReturnsRe
...
+        var result = _controller.Edit(1, editVM);
+        var model = ((ViewResult)result).Model;
+
+        //Assert
+        Assert.IsType<ViewResult>(result);
+        Assert.False(_controller.ModelState.IsValid);
+        Assert.Equal(JsonSerializer.Serialize(editVM), JsonSerializer.Serialize(model));
+        Assert.NotNull(result);
+        Assert.NotNull(model);
+
```
Comment (SamDotNetDev) [inline ParkingZoneApp.Tests/Controller/ParkingSlotControllerTests.cs]: space

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs [hunk 5]
```diff
@@ -61,4 +62,52 @@ public IActionResult Create(CreateViewModel createModel)
...
+    public IActionResult Edit(int? id)
+    {
+        var slot = _slotService.GetById(id);
+        if (slot == null)
+        {
+            return NotFound();
+        }
+        var editModel = new EditViewModel(slot);
+        return View(editModel);
+
```
Comment (SamDotNetDev) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs]: space

File: ParkingZoneApp/ViewModels/ParkingSlotViewModels/EditViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        Id = slot.Id;
+        Number = slot.Number;
+        IsAvailableForBooking = slot.IsAvailableForBooking;
+        ParkingZoneId = slot.ParkingZoneId;
+        Category = slot.Category;
+    }
+    public EditViewModel()
+    {
+
```
Comment (SamDotNetDev) [inline ParkingZoneApp/ViewModels/ParkingSlotViewModels/EditViewModel.cs]: not user Constructor taking place lets make it beautifull public EditViewModel() { }

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs [hunk 7]
```diff
@@ -61,4 +62,52 @@ public IActionResult Create(CreateViewModel createModel)
...
+    }
+
+    // POST: Admin/ParkingSlot/Edit/5
+    [HttpPost]
+    [ValidateAntiForgeryToken]
+    public IActionResult Edit(int id, EditViewModel editModel)
```
Comment (SamDotNetDev) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs]: In this action you are not checking if the submitted ParkingSlot.Number is the same. 
Aytmoqchimanki agar Editda ParkingSlot Numberni uzgartirmasdan Submit qsez Exist db cb qolad uzinikidaqa qomed

**Review by SamDotNetDev — CHANGES_REQUESTED** [review]

**Review by SamDotNetDev — APPROVED** [review]

File: ParkingZoneApp/ViewModels/ParkingSlotViewModels/EditViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
+﻿using ParkingZoneApp.Enums;
+using ParkingZoneApp.Models;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+
+namespace ParkingZoneApp.ViewModels.ParkingSlotViewModels;
+
+public class EditViewModel
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/ViewModels/ParkingSlotViewModels/EditViewModel.cs]: photo_2024-05-10_15-19-39

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs [hunk 9]
```diff
@@ -61,4 +63,50 @@ public IActionResult Create(CreateViewModel createModel)
...
+        {
+            return NotFound();
+        }
+
+        if (ModelState.IsValid)
+        {
+            try
+            {
+                ParkingSlot slot = editModel.MapToModel(editModel);
+                _slotService.Update(slot);
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs]: firstly you are updating slot and then checking this is slot exists with this number ?!

**Review by abdullokhtolibjonov — CHANGES_REQUESTED** [review]

File: ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs (hunk 9 above)
Comment (iamurtazo) [reply ParkingZoneApp/Areas/Admin/Controllers/ParkingSlotController.cs]: Well, I have written it by looking at parkingzone controller which was added scaffold. And it has not caused any problem yet.

File: ParkingZoneApp/ViewModels/ParkingSlot/EditViewModel.cs [hunk 10]
```diff
@@ -0,0 +1,42 @@
+﻿using ParkingZoneApp.Enums;
+using ParkingZoneApp.Models;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+
+namespace ParkingZoneApp.ViewModels.ParkingSlotViewModels;
+
+public class EditViewModel
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/ViewModels/ParkingSlot/EditViewModel.cs]: image

Comment (abdullokhtolibjonov) [reply ParkingZoneApp/ViewModels/ParkingSlot/EditViewModel.cs]: look at naming conventions

**Review by abdullokhtolibjonov — CHANGES_REQUESTED** [review]

**Review by abdullokhtolibjonov — APPROVED** [review]

**Pull request merged**